test/test_shared.cpp

```diff
@@ -4345,4 +4347,67 @@ TEST(Shared_WriteToFail)
     CHECK(*tr == *dest);
 }
 
+NONCONCURRENT_TEST_IF(Shared_LockFileConcurrentInit, testing_supports_spawn_process)
+{
+    auto path = realm::test_util::get_test_path(test_context.get_test_name(), ".test-dir");
+    test_util::TestDirGuard test_dir(path, false);
+    test_dir.do_remove = SpawnedProcess::is_parent();
+    auto lock_prefix = std::string(path) + "/lock";
+
+    struct Mutex : InterprocessMutex {
+        SharedPart sp;
+
+        Mutex(Mutex&&) = default;
+        Mutex& operator=(Mutex&&) = default;
+
+        Mutex(const std::string& name, const std::string& lock_prefix_path)
+        {
+            set_shared_part(sp, lock_prefix_path, name);
+        }
+        ~Mutex()
+        {
+            release_shared_part();
+        }
+    };
+
+    for (size_t i = 0; i < 10; ++i) {
+        std::vector<std::unique_ptr<SpawnedProcess>> spawned;
+
+        // create multiple processes initializing multiple same purpose locks
+        for (size_t j = 0; j < 10; ++j) {
+            spawned.emplace_back(
+                test_util::spawn_process(test_context.test_details.test_name, util::format("child [%1]", i)));
+
+            if (spawned.back()->is_child()) {
```

Comment:
Isn't it timing dependent what spawned.back() refers to when it executes in a child while the parent is still spawning other children?

Reply:
I think I just (mis)read the spawn_process as a fork() system call...